src/streamlink/session.py

```diff
@@ -453,42 +452,20 @@ def load_plugins(self, path):
         """Attempt to load plugins from the path specified.
 
         :param path: full path to a directory where to look for plugins
-
         """
-        for loader, name, ispkg in pkgutil.iter_modules([path]):
-            file, pathname, desc = imp.find_module(name, [path])
-            # set the full plugin module name
-            module_name = "streamlink.plugin.{0}".format(name)
-
+        user_input_requester = self.get_option("user-input-requester")
+        for (loader, name, _) in pkgutil.iter_modules([path]):
             try:
-                self.load_plugin(module_name, file, pathname, desc)
-            except Exception:
+                mod = loader.find_module(name).load_module(name)
+            except:
                 sys.stderr.write("Failed to load plugin {0}:\n".format(name))
                 print_small_exception("load_plugin")
-
                 continue
-
-    def load_plugin(self, name, file, pathname, desc):
-        # Set the global http session for this plugin
-        user_input_requester = self.get_option("user-input-requester")
-        api.http = self.http
-
-        module = imp.load_module(name, file, pathname, desc)
-
-        if hasattr(module, "__plugin__"):
-            module_name = getattr(module, "__name__")
-            plugin_name = module_name.split(".")[-1]  # get the plugin part of the module name
-
-            plugin = getattr(module, "__plugin__")
-            plugin.bind(self, plugin_name, user_input_requester)
-
-            if plugin.module in self.plugins:
-                log.debug("Plugin {0} is being overridden by {1}".format(plugin.module, pathname))
-
-            self.plugins[plugin.module] = plugin
-
-        if file:
-            file.close()
+            if hasattr(mod, "__plugin__") and issubclass(mod.__plugin__, Plugin):
+                mod.__plugin__.bind(self, name, user_input_requester)
+                if mod.__plugin__.module in self.plugins:
+                    log.debug("Plugin {0} is being overridden".format(mod.__plugin__.module))
```

Comment:
are there any reason for removing the path?

Reply:
It wasn’t easily available in the previous iteration, I will add it back :)